text/15_testing_basics.md

```diff
@@ -153,9 +153,9 @@ Consider a desktop email application.  Tasked with testing this, a black box tes
 
 Aspects of the system, such as what algorithm was used or what kind of memory allocation scheme is used, can be inferred by the black-box tester, but their concern is focused on the results of the system running.  For example, a black-box tester might notice that the system gets extremely slow when sorting a user's inbox when the user has thousands of messages.  The reason for this is a poor algorithm choice, but the tester is concerned with reporting the result that it is slow, not looking up or analyzing the algorithm.
 
-One could consider __white box testing__ to be the opposite of black box testing.  In white box testing, the tester has intimate knowledge of the codebase and tests code itself.  White box tests can test individual functions of the code, often looking at much more granular aspects of the system than black box tests.
+In regards to knowledge of the system under test, __white box testing__ is the opposite of black box testing.  In white box testing, the tester has intimate knowledge of the codebase and tests code itself.  White box tests can test individual functions of the code, often looking at much more granular aspects of the system than black box tests.
 
-Continuing the example of a desktop email application, white box tests might check the actual `sort(EmailEntry[] emails)` function, sending in various values to see what the function returns or does.  White box testers would care about what happened specifically if a zero-length array or null reference were passed in, whereas a black box tester would only care about that if they specifically could cause it to happen from the user interface.  White box tests access the code as code - checking that return values from functions are correct, ensuring that objects are instantiated properly, etc. - instead of looking at the system from a user's perspective.
+Continuing the example of a desktop email application, white box tests might check the actual `sort(EmailEntry[] emails)` function, sending in various values to see what the function returns or does.  White box testers would care about what happened specifically if a zero-length array or null reference were passed in, whereas a black box tester would only care about what happens if they attempt to sort an empty list of emails in the application itself.  White box tests access the code as code - checking that return values from functions are correct, ensuring that objects are instantiated properly, etc. - instead of looking at the system from a user's perspective.
```

Comment:
This sentence was confusing for me. I think it's useful here to illustrate what the "edge case" actually looks like in a black box testing perspective